backend/services/portale_alloggi_service.py

```
@@ -0,0 +1,744 @@
+</soap:Envelope>"""
+
+    def _create_auth_headers(self) -> Dict[str, str]:
+        """Create headers for authentication request."""
+        return {
+            'Content-Type': SOAP_CONTENT_TYPE,
+            'SOAPAction': 'AlloggiatiService/GenerateToken'
+        }
+
+    def _parse_auth_response(self, response_text: str) -> Optional[str]:
+        """Parse authentication response and extract token."""
+        root = ET.fromstring(response_text)
```

Ruff (0.12.2)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)